src/pages/postgraphile/jwk-verification.md

```diff
@@ -1,184 +1,205 @@
 # Or:
-$ npm install --save jsonwebtoken
+npm install --save express express-jwt jwks-rsa
 ```
 
+### Prior Knowledge & Context
+
+As a developer, the three essential aspects of Auth0 are: 
+
+- [_APIs_](https://auth0.com/docs/apis) and 
+    [_Applications_](https://auth0.com/docs/applications)
+- [_JWT types_](https://auth0.com/docs/tokens) 
+    (e.g. _ID Token_ vs. _Access Token_)
+- Authentication and Authorization [_Flows_](https://auth0.com/docs/flows) 
+
+To keep it simple, in this guide we will be dealing with an 
+[Access Token](https://auth0.com/docs/tokens/overview-access-tokens) 
+granted by an API which we will need to verify.
+
 ## Getting Started
 
-When setting up your PostGraphile server, the core setting for verifying your
-JWT against a JWK is the [`pgSettings`
-function](/postgraphile/usage-library/#pgsettings-function). The `pgSettings`
-function, in our case, will be an asynchronous function that inspects the
-incoming web request, extracting and verifying the authorization header against
-Auth0.
+You will need two values from your Auth0 configuration: The Auth0 _tenant 
+domain name_, and the API _identifier._ 
 
-Let's add a simple function to `pgSettings` in order to capture the authorization header. If there is no header, we will throw an error, aborting the request.
+```
+const jwt = require('express-jwt');
+const jwksRsa = require('jwks-rsa');
 
-```javascript{3-11}
-app.use(
-  postgraphile(process.env.DATABASE_URL, process.env.DB_SCHEMA, {
-    pgSettings: req => {
-      const authHeader = req.headers.authorization;
-      if (!authHeader) {
-        throw new Error("No authorization header provided.");
-      }
+// ...
+
+// Authentication middleware. When used, the
+// Access Token must exist and be verified against
+// the Auth0 JSON Web Key Set.
+// On successful verification, the payload of the
+// decrypted Access Token is appended to the
+// request (`req`) as a `user` parameter.
+const checkJwt = jwt({
+  // Dynamically provide a signing key
+  // based on the kid in the header and 
+  // the signing keys provided by the JWKS endpoint.
+  secret: jwksRsa.expressJwtSecret({
+    cache: true,
+    rateLimit: true,
+    jwksRequestsPerMinute: 5,
+    jwksUri: `https://YOUR_DOMAIN/.well-known/jwks.json`
+  }),
+
+  // Validate the audience and the issuer.
+  audience: 'YOUR_API_IDENTIFIER',
+  issuer: `https://YOUR_DOMAIN/`,
+  algorithms: ['RS256']
+});
+```
+
+(note: if we were processing an [ID Token](https://auth0.com/docs/tokens/id-token)
+instead of an Access Token, the _audience_ would be the _Client ID_ instead)
+
+Remember that a JWT has [three _period-separated_ sections](https://jwt.io/introduction/): header, payload,
+and signature. On successful verification, the payload will be available for
+us to save inside the Postgraphile request via the
+[`pgSettings`](https://www.graphile.org/postgraphile/usage-library/#exposing-http-request-data-to-postgresql)
+object.
```

Comment:
This should maybe be “function” or “return value”?

Reply:
Agreed, changed to 'function'